packages/api/src/beacon/routes/beacon/pool.ts

```diff
@@ -236,49 +316,55 @@ export function getDefinitions(config: ChainForkConfig): RouteDefinitions<Endpoi
     submitPoolAttestations: {
       url: "/eth/v1/beacon/pool/attestations",
       method: "POST",
+      req: {
+        writeReqJson: ({signedAttestations}) => ({
+          body: AttestationListTypePhase0.toJson(signedAttestations),
+        }),
+        parseReqJson: ({body}) => ({signedAttestations: AttestationListTypePhase0.fromJson(body)}),
+        writeReqSsz: ({signedAttestations}) => ({body: AttestationListTypePhase0.serialize(signedAttestations)}),
+        parseReqSsz: ({body}) => ({signedAttestations: AttestationListTypePhase0.deserialize(body)}),
+        schema: {
+          body: Schema.ObjectArray,
+        },
+      },
+      resp: EmptyResponseCodec,
+    },
+    submitPoolAttestationsV2: {
+      url: "/eth/v2/beacon/pool/attestations",
+      method: "POST",
       req: {
         writeReqJson: ({signedAttestations}) => {
-          const fork = config.getForkName(signedAttestations[0].data.slot);
+          const fork = config.getForkName(signedAttestations[0]?.data.slot ?? 0);
```

Comment:
In practice, we won't be calling this api with an empty array but it's possible based on the types and the spec does not explicitly state that the array is not allowed to be empty.

We don't have access to the clock here, so it's hard as far as I can see to determine the current fork but passing 0 (translating to phase0) should be fine as well since array is empty anyways.